src/dolphin/iniFile.ts

```diff
@@ -0,0 +1,278 @@
+  }
+
+  /** Differs from IniFile.cpp via:
+   * Instead of editing keyOut and valueOut by reference, return them */
+  public parseLine(line: string): Array<string | null> {
+    let retValueOut: string | null = null;
+    let keyOut: string | null = null;
+
+    if (line === "" || line[0] === "#") {
+      return [null, null];
+    }
+
+    const firstEquals = line.indexOf("=");
+    if (firstEquals !== -1) {
+      keyOut = line.substring(0, firstEquals).replace(/\s+/g, "");
+      retValueOut = line
+        .substring(firstEquals + 1)
+        .replace(/\s+/g, "")
+        .replace(/['"]+/g, "");
+    }
+
+    return [keyOut, retValueOut];
+  }
+
+  /**Differs from IniFile.cpp by:
+   * returns section object, not pointer
+   */
+  public getSection(section_name: string): Section | undefined {
+    const section = this.sections.find((section) => section.name === section_name);
+    return section;
+  }
+
+  /**Differs from IniFile.cpp by:
+   * returns section object, not pointer
+   */
+  public getOrCreateSection(section_name: string): Section {
+    let section = this.getSection(section_name);
+    if (section === undefined) {
+      section = new Section(section_name);
+      this.sections.push(section);
+    }
+    return section;
+  }
+
+  public deleteSection(section_name: string): boolean {
+    const s = this.getSection(section_name);
+    if (s === undefined) {
+      return false;
+    }
+    this.sections.splice(this.sections.indexOf(s), 1);
+    return true;
+  }
+
+  public exists(section_name: string): boolean {
+    return this.getSection(section_name) != undefined;
+  }
+
+  public setLines(section_name: string, lines: string[]): void {
+    const section = this.getOrCreateSection(section_name);
+    section.setLines(lines);
+  }
+
+  public deleteKey(section_name: string, key: string): boolean {
+    const section = this.getSection(section_name);
+    if (section === undefined) {
+      return false;
+    }
+    return section.delete(key);
+  }
+
+  /**Differs from IniFile.cpp by:
+   * returns keys instead of passing it by reference
+   */
+  public getKeys(section_name: string): string[] | boolean {
+    const section = this.getSection(section_name);
+    if (section === undefined) {
+      return false;
+    }
+    return section.keys_order;
+  }
+
+  /**Differs from IniFile.cpp by:
+   * returns lines instead of passing it by reference
+   */
+  public getLines(section_name: string, remove_comments = false): string[] {
+    const section = this.getSection(section_name);
+    if (section === undefined) {
+      return [];
+    }
+
+    const lines = section.getLines(remove_comments);
+
+    return lines;
+  }
+
+  public async load(fileName: string, keep_current_data = true): Promise<boolean> {
+    if (!keep_current_data) {
+      this.sections = [];
+    }
+
+    const ins = fs.createReadStream(fileName);
+    ins.on("error", (e) => {
+      console.log("failed to read file with error", e);
+      alert(e);
```

Comment:
```suggestion
      log.error("failed to read file:", e);
```

We really shouldn't be using `alert` for error handling.